pkg/apis/run/v1alpha1/runstatus_types.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// This package exists to avoid an import cycle between v1alpha1 and v1beta1.
+// It contains common definitions needed by v1alpha1.Run and v1beta1.PipelineRun.
```

Comment:
I don't think this is required **and** I think we should move **all** types related to `Run` types, not just the one that did pause problem for import cycle.

Reply:
This means `Run`, `RunSpec`, `RunList`, … types

Reply:
> We should move all Run related types to pkg/apis/run/v1alpha1 instead of just the one that were a problem for import cycle. This would allow us to still not import anything from "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1" but "github.com/tektoncd/pipeline/pkg/apis/run/v1alpha1" instead — similar to what we do with pkg/apis/resource/v1alpha1

Okay, so I should not even leave behind any aliases in "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"?  In other words it's a completely incompatible change of the package.  It doesn't matter to me, I'll do whatever everyone wants here.

Reply:
Our policy is that we don't care about breaking changes that only affect Go clients. This is especially true for anything related to Runs, which are alpha and haven't had much real-world usage outside of, honestly, you or me. 😆

Reply:
@GregDritschler yeah I think it's fine. The alias can stay there if we feel there is a need for them there… but I am fine if they are not there any more, as @ImJasonH said, we don't care too much about breaking changes that only affect Go clients, **especially** for `v1alpha` :upside_down_face:

Reply:
Moving the RunSpec is harder because it uses v1beta1.Params and v1alpha1.TaskRef.  This means a lot of refactoring which must be done carefully to avoid breaking these widely used types.  I discussed offline with Vincent and he agreed this could be considered for a follow on to this PR.